### PR TITLE
Add clang-tidy external examples

### PR DESCRIPTION
**Pull request by MichelleCDjunaidi**
Motivation: Clang has a page where they list out external examples: https://clang.llvm.org/docs/ExternalClangExamples.html. While I was writing an out-of-tree clang-tidy plugin, I wanted such a page to exist also as it would have helped development. The example repo listed in the proposed page has been useful for me.

### CODE REVIEW

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (5chmidti) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:17]: What do you think about:
> ..., please share it on <discourse> for wider visibility, and open a pull-request to have it added here. ~(or if you are already a clang-tidy contributor, feel free to directly commit additions)~

**Review by 5chmidti — COMMENTED** [review]
Great idea to add this. Although, we should be careful to not end up with abandoned projects. I sampled three from the page in Clang, and they had their last activity at around ~10 years. At that point, API's have changed. Either way, adding external examples is a good idea IMO
(I should clean up my external clang-tidy template that's still private :) )

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (MichelleCDjunaidi) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:17]: LGTM, will do

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (5chmidti) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:8]: unintended double backtick

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (5chmidti) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:17]: Maybe instead of `..., generally they must have...` 
->

`..., the listed projects should be open-source.`
or
`..., they generally must have code available`
or
`..., they should have code available.`

**Comment by MichelleCDjunaidi** [comment]
Side note, but I don't have write permissions, so once the PR looks great feel free to land it on my behalf.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (EugeneZelenko) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:1]: Please make same length as title. Same below.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (EugeneZelenko) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:8]: :program:`clang-tidy`. Same in other places.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (EugeneZelenko) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:13]:
```suggestion
If you know of (or wrote!) a tool or project using Clang-tidy, please share it 
```

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (EugeneZelenko) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:32]:
```suggestion
    "This folder contains Clang-tidy plugins."
```

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (MichelleCDjunaidi) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:13]: That one is directly quoted from the repo's README.md, as https://clang.llvm.org/docs/ExternalClangExamples.html also directly quotes each repo's README.md. Is it fine for us to alter that?

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (MichelleCDjunaidi) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:32]: That one is directly quoted from the repo's README.md, as https://clang.llvm.org/docs/ExternalClangExamples.html also directly quotes each repo's README.md. Is it fine for us to alter that?

Comment (EugeneZelenko) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:32]: `clang-tools-extra/docs/clang-tidy/index.rst` is better reference. But it is predominantly `Clang-Tidy` there.

**Comment by MichelleCDjunaidi** [comment]
@EugeneZelenko @5chmidti Thoughts on the fixed formatting? I followed index.rst convention.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (MichelleCDjunaidi) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst]: followed ``index.rst`` conventions.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (MichelleCDjunaidi) [reply clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:26]: Most likely so. Will remove.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (SimplyDanny) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:30]: Please and an empty newline.

**Review by SimplyDanny — APPROVED** [review]
One more nit. Otherwise looks good to me.

File: clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst
Comment (5chmidti) [inline clang-tools-extra/docs/clang-tidy/ExternalClang-TidyExamples.rst:21]: `is using, for example, the AST Matchers and diagnostics of Cang,` sounds better

**Review by 5chmidti — APPROVED** [review]

**Comment by MichelleCDjunaidi** [comment]
I will need help to merge this once approved. Thank you so much!

**Comment by MichelleCDjunaidi** [comment]
@5chmidti @SimplyDanny can anyone help merge this into main?

**Comment by SimplyDanny** [comment]
There is an error in doc generation. Is it caused by the new file being added? Is there a configuration that needs adaption?

**Comment by MichelleCDjunaidi** [comment]
I asked in the forums. https://discourse.llvm.org/t/fix-error-in-upstream-doc-generation-due-to-addition-of-new-section/81368. Hopefully someone knows what to add...

**Comment by HerrCai0907** [comment]
You can add `toctree` in `clang-tools-extra/docs/clang-tidy/index.rst`
example commit: https://reviews.llvm.org/D54945

**Comment by MichelleCDjunaidi** [comment]
Well, the documentation builds now thanks to @HerrCai0907. Does it look fine @SimplyDanny @5chmidti ?

**Pull request merged**
